CONTRIBUTING.md

```diff
@@ -8,144 +8,189 @@ The following is a short set of guidelines for contributing to Iroha.
 
+While logs printed in the command's `stdout` are sufficient, you may find it more convenient to produce `json`-formatted logs into a separate file and parse them using either
 - [node-bunyan](https://www.npmjs.com/package/bunyan)
 - [rust-bunyan](https://crates.io/crates/bunyan)
 
-## Styleguides
+Set the `LOG_FILE_PATH` environment variable to an appropriate location to store the logs and parse them using the above packages.
+
+## Style Guides
 
 ### Git Style Guide
 
--  **Sign-off every commit** with [DCO](https://github.com/apps/dco):
-   `Signed-off-by: $NAME <$EMAIL>`. You can do it automatically using
-   `git commit -s`
--  **Use present tense** ("Add feature", not "Added feature").
--  **Use imperative mood** ("Deploy docker to..." not "Deploys docker
-   to...").
--  Write meaningful commit message.
--  Limit the first line of commit message to 50 characters or less
--  First line of commit message must contain summary of work done,
-   second line must contain empty line, third and other lines can
-   contain list of commit changes
--  Use [Git Rebase Workflow](https://git-rebase.io/)
+-  We require you to **Sign-off every commit**. If you don't [DCO](https://github.com/apps/dco) will not let you merge. Please add: `Signed-off-by: $NAME <$EMAIL>` as the final line of your commit message. You can do it automatically using `git commit -s`
+-  It's also useful to format your commit messages appropriately. Try to keep them short.
+-  **Prefer present tense** ("Add feature", not "Added feature").
+-  **Prefer imperative mood** ("Deploy to docker..." not "Deploys to docker...").
+-  Write a meaningful commit message. Imagine that you're looking at someone's code, titled "Fixed error".
+-  Limit the first line of your commit message to 50 characters or less.
+-  The first line of your commit message should contain the summary of the work you've done. If you need more than one line, leave a blank line between each pargraph and describe your changes in the middle. The last line must be the sign-off.
+-  Use the [Git Rebase Workflow](https://git-rebase.io/). Avoid using `git pull` use `git pull --rebase` instead.
 
 
 ### Rust Style Guide
 
-- Use `cargo fmt --all`
-- Prefer using `mod.rs` inside the module directory to the `.rs` file named the same as the module in the top level directory.
-- Use domain-first modules structure. For example `domain::isi::*`. Such a way
-complex uses will be easier to incude in dependent modules.
-- Do not use whitespaces or empty lines inside function bodies.
-- Put public methods first in your impl blocks.
-- Put inner modules after `self` module content, but before `tests` module.
```

Comment:
we don't want this anymore?

Reply:
Easy. The re-ordering of modules is handled by `cargo fmt`, `tests` at the end is standard practice in `rust` code. Whitespace in `fn` bodies is not always bad.

Reply:
I was just referring to the last item, but ok